scipy/stats/_binned_statistic.py

```diff
@@ -686,11 +686,12 @@ def _bin_numbers(sample, nbin, edges, dedges):
     # For the rightmost bin, we want values equal to the right
     # edge to be counted in the last bin, and not as an outlier.
     for i in xrange(Ndim):
-        # Find the rounding precision
-        try:
-            decimal = int(-np.log10(dedges[i].min())) + 6
-        except RuntimeWarning:
-            raise ValueError('The smallest edge difference is numerically 0.')
+        with np.errstate(divide='raise'):
+            # Find the rounding precision
+            try:
+                decimal = int(-np.log10(dedges[i].min())) + 6
+            except Exception:
```

Comment:
Catching a generic `Exception` is best avoided, as you likely know. No other choice here?

Reply:
Perhaps the same fix as here?

https://github.com/scipy/scipy/pull/10930/files#diff-9b2d6b68348fecf8d09990be924c77e9R690-R692

Reply:
(well maybe that plus the `FloatingPointError` that NumPy might also raise)

Reply:
in the rebase I added another commit w/the tuple approach.